src/style.js

```diff
@@ -314,19 +315,35 @@ export function applyIndirectStyles(selection, mark, scales, dimensions) {
   applyAttr(selection, "shape-rendering", mark.shapeRendering);
   applyAttr(selection, "paint-order", mark.paintOrder);
   applyAttr(selection, "pointer-events", mark.pointerEvents);
-  if (mark.clip === "frame") {
-    const {x, y} = scales;
-    const {width, height, marginLeft, marginRight, marginTop, marginBottom} = dimensions;
-    const id = getClipId();
-    selection
-      .attr("clip-path", `url(#${id})`)
-      .append("clipPath")
-      .attr("id", id)
-      .append("rect")
-      .attr("x", marginLeft - (x?.bandwidth ? x.bandwidth() / 2 : 0))
-      .attr("y", marginTop - (y?.bandwidth ? y.bandwidth() / 2 : 0))
-      .attr("width", width - marginRight - marginLeft)
-      .attr("height", height - marginTop - marginBottom);
+  switch (mark.clip) {
+    case "frame": {
+      const {x, y} = scales;
+      const {width, height, marginLeft, marginRight, marginTop, marginBottom} = dimensions;
+      const id = getClipId();
+      selection
+        .attr("clip-path", `url(#${id})`)
+        .append("clipPath")
+        .attr("id", id)
+        .append("rect")
+        .attr("x", marginLeft - (x?.bandwidth ? x.bandwidth() / 2 : 0))
+        .attr("y", marginTop - (y?.bandwidth ? y.bandwidth() / 2 : 0))
+        .attr("width", width - marginRight - marginLeft)
+        .attr("height", height - marginTop - marginBottom);
+      break;
+    }
+    case "sphere": {
+      const {projection} = context;
+      if (projection == null) break;
+      const path = geoPath(projection);
+      const id = getClipId();
+      selection
+        .attr("clip-path", `url(#${id})`)
+        .append("clipPath")
+        .attr("id", id)
+        .append("path")
+        .attr("d", path({type: "Sphere"}));
```

Comment:
This makes me wonder if you could say `clip: {type: "Sphere"}` to pass in arbitrary geometry. But I like starting simple like this for now.